src/main/uploader.ts

```diff
@@ -103,23 +119,27 @@ export class MinidumpUploader {
    * @returns A promise that resolves when the upload is complete.
    */
   public async uploadMinidump(request: MinidumpRequest): Promise<void> {
-    logger.log('Uploading minidump', request.path);
+    if (typeof (this._transport as any).sendRequest !== 'function') {
+      logger.warn("Your transport doesn't implement sendRequest");
+      logger.warn('Skipping sending minidump');
+      return;
+    }
+    logger.log('Sending minidump', request.path);
 
+    const transport = this._transport as NetTransport;
     try {
-      const body = new FormData();
-      body.append('upload_file_minidump', fs.createReadStream(request.path));
-      body.append('sentry', JSON.stringify(request.event));
-      const response = await fetch(this._url, { method: 'POST', body });
+      const requestForTransport = await this._toMinidumpRequest(request.event, request.path);
+      const response = await transport.sendRequest(requestForTransport);
 
       // Too many requests, so we queue the event and send it later
-      if (response.status === CODE_RETRY) {
+      if (response.status === Status.RateLimit) {
```

Comment:
Here, we have to make two changes:
1. On a `429` response, we should never retry. 
2. On top of that, the envelope endpoint may return a `429`, even if the event was accepted. Please check the corresponding `X-Sentry-Rate-Limits` header if a rate limit for the `"error"` category is supported, and ignore the ones for attachments for now.

We can later follow up with a more nuanced rate limiting logic.